evident/stats.py

```diff
@@ -72,3 +74,96 @@ def calculate_cohens_f(*arrays) -> float:
     effect_size_numerator = np.sqrt(effect_size_numerator)
 
     return effect_size_numerator/pooled_std
+
+
+def calculate_eta_squared(data: pd.DataFrame) -> float:
+    """Calculate eta squared for repeated measures ANOVA.
+
+    tinyurl.com/ycktzet3
+
+    eta2 = SS_cond / (SS_cond + SS_error)
+    SS_error = SS_within - SS_subj
+    SS_cond = sum_i^k (n_i(x_i - x)^2
+    SS_within = sum(xi_1 - x1)^2 + sum(xi2 - x2)^2 ... + sum(xik - xk)^2
+
+    :param data: Repeated measures data. Index is each subject, columns are
+        each group. Data should have no missing values.
+    :type data: pd.DataFrame
+
+    :returns: Effect size as eta_squared
+    :rtype: float
+    """
+    if data.isna().any().any():
+        raise ValueError(
+            "Cannot calculate effect size of repeated measures with missing "
+            "values."
+        )
+    k = data.shape[1]  # number of groups
+    mu_total = np.mean(data.values.ravel())
+
+    ss_within = data.apply(
+        lambda x: np.power(x - np.mean(x), 2),
+        axis=0
+    ).sum().sum()
+
+    ss_subj = data.apply(
+        lambda x: np.power(np.mean(x) - mu_total, 2),
+        axis=1
+    ).sum() * k
+
+    subjs_per_cond = (~data.isna()).sum(axis=0)
```

Comment:
aren't we assured that none of the data are na?

Reply:
Yup, good catch - vestige of a previous version of the function.